wazo_debug/access.py

```diff
@@ -0,0 +1,138 @@
+        parser.add_argument(
+            '-s',
+            '--remote-server',
+            action='store',
+            help='The remote server where the tunnel will be opened.',
+        )
+        parser.add_argument(
+            '-r',
+            '--remote-server-port',
+            action='store',
+            help='The remote server port.',
+        )
+        parser.add_argument(
+            '-u',
+            '--remote-user',
+            action='store',
+            help='The unix user on the remote server.',
+        )
+        parser.add_argument(
+            '-i',
+            '--identity',
+            action='store',
+            help="The remote user's SSH private key file.",
+        )
+        parser.add_argument(
+            '-t',
+            '--timeout',
+            action='store',
+            help='The tunnel will timeout after this much seconds. Defaults to 8 hours.',
+            default=28800,
+        )
+        return parser
+
+    def take_action(self, parsed_args):
+
+        # We need a retry mechanism just in case there's a port collision on
+        # the remote server, in which case we make a new attempt with a different
+        # random port
+        for attempt in range(ssh_max_retries):
+            result = self.open_access(parsed_args)
+            if result == 0:
+                break
+        else:
+            logger.critical('Max retries (%s) exceeded, exiting.', ssh_max_retries)
+            exit(1)
+
+    def open_access(self, parsed_args):
+        remote_port = random.randint(22022, 22222)
+
+        ssh_command = [
+            'ssh',
+            '-o',
+            'StrictHostKeyChecking=no',
+            '-o',
+            'PreferredAuthentications=publickey',
+            '-o',
+            'ExitOnForwardFailure=yes',
+            '-o',
+            'ServerAliveInterval=15',
+            '-o',
+            'ServerAliveCountMax=10',
+            '-o',
+            'ControlMaster=no',
+            '-l',
+            parsed_args.remote_user,
+            '-p',
+            str(parsed_args.remote_server_port),
+            '-i',
+            parsed_args.identity,
+            '-R',
+            f'0.0.0.0:{remote_port}:localhost:{parsed_args.local_port}',
+            parsed_args.remote_server,
+            'sleep infinity',
+        ]
+
+        full_command = ['timeout', str(parsed_args.timeout)] + ssh_command
+
+        human_readable_timeout = time.strftime(
+            "%H:%M:%S", time.gmtime(parsed_args.timeout)
+        )
+
+        logger.debug(
+            f'''\
+Opening the access using this command:
+    {' '.join(full_command)}'''
+        )
+
+        logger.info(
+            f'''\
+Access will open now. To connect as the 'root' user on the current server, one can use the following command:
+
+    ssh root@{parsed_args.remote_server} -p {remote_port}
+
+Keep the terminal open and it'll last for {human_readable_timeout}. Close the terminal or hit Ctrl-C to cut it.'''
+        )
```

Comment:
Avoid format string in logger, but no objection to keep for clarity

Reply:
Agreed, I kept them for clarity.